test.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const request = require('request')
+const Fastify = require('fastify')
+const fs = require('fs')
+
+test('reply.view exist', t => {
+  t.plan(6)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {
+    engine: {
+      ejs: require('ejs')
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    t.ok(reply.view)
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
+      method: 'GET',
+      uri: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(response.headers['content-length'], '' + body.length)
+      t.deepEqual(JSON.parse(body), { hello: 'world' })
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view should return 500 if some data is missing', t => {
+  t.plan(3)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {
+    engine: {
+      ejs: require('ejs')
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view()
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
+      method: 'GET',
+      uri: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 500)
+      fastify.close()
+    })
+  })
+})
+
+/* test('register callback should throw if the engine is missing', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {}, err => {
+    t.error(err)
+    t.is(err.message, 'Missing engine')
+    t.pass()
+  })
+}) */
```

Comment:
the problem is that `t.error` does not do what you think it's doing. `t.error` fails if `err` exists. You should replace it with `t.ok(err)`